Libplanet.Tests/Blockchain/BlockChainTest.cs

```diff
@@ -769,19 +790,35 @@ public void ExecuteActions()
                 { MinerReward.RewardRecordAddress, (Text)$"{minerAddress},{minerAddress}" },
             };
 
-            Log.Debug("{@actual}", _blockChain.Store.GetBlockStates(block1.Hash));
+            if (_blockChain.StateStore is IBlockStatesStore blockStatesStore)
+            {
+                Log.Debug("{@actual}", blockStatesStore.GetBlockStates(block1.Hash));
+            }
+            else
+            {
+                // NOTE: The reason to throw exception is, it checks only block states and
+                //       block state is a feature of IBlockStateStore. So if _blockChain.StateStore
+                //       became use IStateStore implementation not to implement IBlockStateStore,
+                //       then this test should be updated also.
```

Comment:
Sorry, but could you elaborate this again?

Reply:
(@moreal explained it well on the chat; thanks!)

The reason why we need [type testing][1] here is we coders are aware that `_blockChain.StoreState` is `DefaultStoreFixture.StateStore` and again `DefaultStoreFixture.StateStore` is just a `DefaultStore` casted to `IStateStore`, but the C# compiler isn't aware that.  If a problem is the compiler is unaware of it… let it know!

The current way to initialize `_blockChain` is like below:

https://github.com/planetarium/libplanet/blob/7e2e670e3b0ba06cdc399b87067b1ec4ce8fd7a5/Libplanet.Tests/Blockchain/BlockChainTest.cs#L42-L48

So we could use `_fx.BlockStatesStore` instead of  `_blockChain.StateStore` here, right?

[1]: https://softwareengineering.stackexchange.com/questions/253704/when-is-type-testing-ok

Reply:
You're right. Also if we became to use `_fx.BlockStatesStore` and `_blockChain.StoreState` became not to use an implementation of `IBlockStatesStore`, it will raise `NullReferenceException` and the test will fail. Thanks! 🙇